### DISCUSSION TITLE
Provide packages and how to install?

### DISCUSSION THREAD

**Discussion by LinuxWOW** [linuxdeepin/developer-center#3225] [Features Request & Ideas | 特性请求 & 头脑风暴]
I would like the logout from deepin if possible on Ubuntu or other OS. Where the packege SRC file. It would be nice if Deepin provided packeges and how to install, and also not linked to deepin. I was looking to get greeter as well but didn't know how to install it becuase of no package lack of information  on Deepin to build?

**Comment by BLumia** [comment]
Sorry I couldn't understand what you mean...

> I would like the logout from deepin if possible on Ubuntu or other OS.

Do you mean you want to port deepin/DDE packages to Ubuntu or other distro?

> Where the packege SRC file.

If you mean the source code, then all the repositories under [this organization](https://github.com/linuxdeepin/) are the source files. Also, if you want to get the exact same source of the package you use in the deepin distro, you can also use `apt source` to get the source packages.

> ...provided packeges and how to install

We do provide packages for deepin. We also provide some instruction manual for how to build a specific project (although some projects miss a valid README.md file that provide the instruction, in that case, please feel free to ask us if you need any help).

> also not linked to deepin

Do you mean you want us provide 1st-party packages for non-deepin distro?

> I was looking to get greeter as well

For the LightDM greeter, you can build the dde-session-shell package. It's a regular CMake project so regular CMake building steps will work, you can also see how Arch Linux done the building for this package at [here](https://github.com/archlinux/svntogit-community/blob/5b8562b42f55ee5cc7a61eb564b250306d16f9ba/trunk/PKGBUILD#L29-L33). Also, if you already get DDE installed in the deepin distro, you can use `dpkg -S` to find a installed file belongs to which package, and then get related information with the package name.

> but didn't know how to install it becuase of no package lack of information on Deepin to build?

We also plan to improve the documentation of both the deepin distro, and DDE related package and information. For now, if you need any specific help, please feel free to ask us and we're happy to help you.

**Comment by LinuxWOW** [comment]
Im lookin to get logout packege in ubuntu or arch and the greeter for lightdm where is the packege new one from Deepin. also looking to use is without deepin desktop with KDE if able to. There are no packeges for logout or greeter?

**Reply by BLumia** [reply]
I still don't understand what package you are referring to. If you means the repo/package that provides the lightdm greeter which DDE is using, then you're looking for [dde-session-shell](https://github.com/linuxdeepin/dde-session-shell), and you will need to build it manually from source to use it under other distro like Ubuntu since DDE packages are not yet in their official package repository. As for Arch, the package that provides the DDE lightdm greeter is [deepin-session-shell](https://archlinux.org/packages/community/x86_64/deepin-session-shell/).

**Comment by LinuxWOW** [comment]
@BLumia I'm talking about the deepin greeter and the logout, lock packages where are they? How do you build them under different environments, since there are no instructions.

**Comment by LinuxWOW** [comment]
@BLumia i see its in this repo. [https://github.com/linuxdeepin/dde-session-shell](url) and folder src has greeter. But how do you build doesn't give you anything, Do i need to build other deepin packages with it, What do i do?

**Reply by BLumia** [reply]
It's a regular CMake project, regular CMake steps will do. (`mkdir build && cd build`, `cmake ../`, `make` and then `sudo make install`).

If you want to port it to another distro, since each distro may have a different method to create their native package, you need to consult related manual to learn how to create a package. For debian based distro, a `debian` folder already provided by the source repository so you can use it directly if you like.
